qt_ui/windows/mission/flight/settings/FlightAirfieldDisplay.py

```diff
@@ -53,8 +54,11 @@ def __init__(self, game: Game, package_model: PackageModel, flight: Flight) -> N
         self.setLayout(layout)
 
     def update_departure_time(self) -> None:
-        estimator = TotEstimator(self.package_model.package)
-        delay = estimator.mission_start_time(self.flight)
+        if len(self.package_model.package.flights) > 0:
```

Comment:
How could this ever be false? What package does `self.flight` belong to?

Reply:
No package, that's the whole point... It's an exception that occurs when a package is cancelled before creation. Took me a while to track down the program-flow, and I conclude this to be the best place to catch the special case.

Reply:
Actually, I missed a case... This is a 2-in-1 solution for the following situation:

- The scenario as explained in the issue, having opened the edit-flight window, closing it and cancelling the package
- The same as the first scenario, but instead of cancelling the entire package at the end, cancel/abort the <ins>last flight</ins> (it should be the last one since there is only one if you follow the steps provided). So in other words, if you delete the last flight in a "QNewPackageDialog", you get the same exception, provided that you've opened an edit-flight window. In case you have more flights, it doesn't matter which edit-flight window you opened, nor if the flight in question was already deleted.

So I think I can already guess your next question: why is `FlightAirfieldDisplay` still alive after closing the edit-flight window? I believe that's due to `dialogs.py` having a class attribute `QEditFlightDialog` which uses `FlightAirfieldDisplay` implicitly. So looking for a solution in that direction would probably be a lot more complicated.

Upon reviewing this again I do notice that the "else" part is actually unnecessary since the flight in question is being deleted. Just taking the original code and nesting it within the check should be enough.

This brings me to the following though: it might be more interesting to add a boolean value to all flights, indicating whether or not they were cancelled for the sake of readability. Though there ought to be made a distinction between cancelled and aborted. If the flight is already airborne, it shouldn't put this boolean value to true. Does that make sense?

Anyway, let me know which solution you prefer...

Reply:
Actually, I missed a case... This is a 2-in-1 solution for the following situation:

- The scenario as explained in the issue, having opened the edit-flight window, closing it and cancelling the package
- The same as the first scenario, but instead of cancelling the entire package at the end, cancel/abort the <ins>last flight</ins> (it should be the last one since there is only one if you follow the steps provided). So in other words, if you delete the last flight in a "QNewPackageDialog", you get the same exception, provided that you've opened an edit-flight window. In case you have more flights, it doesn't matter which edit-flight window you opened, nor if the flight in question was already deleted.

So I think I can already guess your next question: why is `FlightAirfieldDisplay` still alive after closing the edit-flight window? I believe that's due to `dialogs.py` having a class attribute `QEditFlightDialog` which uses `FlightAirfieldDisplay` implicitly. So looking for a solution in that direction would probably be a lot more complicated.

Upon reviewing this again I do notice that the "else" part is actually unnecessary since the flight in question is being deleted. Just taking the original code and nesting it within the check should be enough.

This brings me to the following though: it might be more interesting to add a boolean value to all flights, indicating whether or not they were cancelled for the sake of readability. Though there ought to be made a distinction between cancelled and aborted. If the flight is already airborne, it shouldn't put this boolean value to true. Does that make sense?

Anyway, let me know which solution you prefer...